src/Sulu/Bundle/MediaBundle/Entity/MediaRepository.php

```diff
@@ -445,6 +445,30 @@ private function getIds(
         return $subQuery->getScalarResult();
     }
 
+    /**
+     * @return array<array{id: int, resourceKey: string, depth: int}>
+     */
+    public function findMediaResourcesOfRootCollection(int $rootCollectionId): array
```

Comment:
not sure what would be better word for `root` here.. maybe `findMediaResourcesByAncestorCollection`

Reply:
or call the method `findMediaResourcesByCollection` and add a `includeDescendantCollections = true` parameter?